examples/provider/provider.tf

```diff
@@ -7,7 +7,7 @@ terraform {
   required_providers {
     checkly = {
       source = "checkly/checkly"
-      version = "1.4.3"
+      version = "1.6.10"
```

Comment:
If we want to add `retry_strategy` to the documentation examples, we should also bump the checkly provider version here. 

This is a big bump, though, and we might have had breaking changes since then. It's probably worth checking that the examples are actually working.